src/Models/page-combined-events-list.php

```diff
@@ -0,0 +1,145 @@
+     */
+    public function description() : ?string {
+        return get_field( 'description' );
+    }
+
+    /**
+     * Get no results text
+     *
+     * @return string
+     */
+    public function no_results() : string {
+        return __( 'No results', 'tms-theme-base' );
+    }
+
+    /**
+     * Get events
+     */
+    public function events() : ?array {
+        try {
+            $response = $this->get_events();
+
+            return $response['events'] ?? [];
+        }
+        catch ( Exception $e ) {
+            ( new Logger() )->error( $e->getMessage(), $e->getTrace() );
+        }
+
+        return null;
+    }
+
+    /**
+     * Get events.
+     *
+     * @return array
+     */
+    protected function get_events() : array {
+        $params = [
+            'keyword'     => get_field( 'keyword' ),
+            'page_size'   => 200, // Use an arbitrary limit as a sanity check.
+            'show_images' => get_field( 'show_images' ),
+            'start'       => 'today',
+            'end'         => '',
+            'location'    => '',
+            'publisher'   => '',
+            'sort'        => '',
+            'text'        => '',
+            'page'        => 1,
+        ];
+
+        $formatter         = new EventsFormatter();
+        $params            = $formatter->format_query_params( $params );
+        $params['include'] = 'organization,location,keywords';
+
+        $cache_group = 'page-combined-events-list';
+        $cache_key   = md5( wp_json_encode( $params ) );
+        $response    = wp_cache_get( $cache_key, $cache_group );
+
+        if ( empty( $response ) ) {
+            $response           = $this->do_get_events( $params );
+            $response['events'] = array_merge( $response['events'], $this->get_manual_events() );
+
+            // Sort events by start datetime objects.
+            usort( $response['events'], function( $a, $b ) {
+                return $a['start_date_raw'] <=> $b['start_date_raw'];
+            } );
+
+            if ( ! empty( $response ) ) {
+                wp_cache_set(
+                    $cache_key,
+                    $response,
+                    $cache_group,
+                    MINUTE_IN_SECONDS * 15
+                );
+            }
+        }
+
+        return $response;
+    }
+
+    /**
+     * Get manual events.
+     *
+     * @return array
+     */
+    protected function get_manual_events() : array {
+        $args = [
+            'post_type'      => PostType\ManualEvent::SLUG,
+            'posts_per_page' => 200, // phpcs:ignore
+            'meta_query'     => [
+                [
+                    'key'     => 'start_datetime',
+                    'value'   => date( 'Y-m-d' ),
+                    'compare' => '>=',
+                    'type'    => 'DATE',
+                ],
+            ],
+        ];
+
+        $query = new WP_Query( $args );
+
+        $events = array_map( function ( $e ) {
+            $id           = $e->ID;
```

Comment:
Nimeisin pidemmin selkeyden vuoks ja ois enemmä standardin näköstä
```suggestion
            $post_id           = $event->ID;
```